extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -107,6 +107,15 @@ export async function promptImportLgtmDatabase(
   return;
 }
 
+export async function retrieveCanonicalRepoName(lgtmUrl: string) {
+  const givenRepoName = parseLgtmUrl(lgtmUrl);
+  const repo = await fetch(`https://api.github.com/repos/${givenRepoName}`).then(res => res.json());
```

Comment:
What happens if there is an error here and something besides `200` is returned? Can you take a look at `checkForFailingResponse` in this file? It would be best to to generalize this function to handle this case as well. You could add a new parameter that would replace the error message `Error downloading database.`.